lib/salus/scanners/base.rb

```diff
@@ -14,6 +14,18 @@ class InvalidScannerInvocationError < StandardError; end
 
     attr_reader :report
 
+    SCANNER_VERSIONS = {
+      'Bandit' => '1.6.2',
+      'Brakeman' => '4.9.1',
+      'BundleAudit' => '0.6.1',
+      'CargoAudit' => '0.12.0',
+      'Gosec' => '2.4.0',
+      'NPMAudit' => '6.14.8',
+      'PatternSearch' => '0.9.0',
+      'Semgrep' => '0.14.0',
+      'YarnAudit' => '1.22.0'
+    }.freeze
```

Comment:
Thoughts on pulling these in from the shell (or gem env) to report the version that will run?

```
`brakeman --version`
=> "brakeman 4.9.1"
cargo audit --version`
=> "cargo-audit 0.12.0"
```
The ones that have gems we can pull dynamically as well:
```
Gem.loaded_specs["brakeman"].version.to_s
=> "4.9.1"
```


etc.  Right now we define the version in the docker container or gem file.  If we bump the docker image but forget to bump the version here we might end up with some mis-versioned reports.

On the base scanner we might want to define a 

```
def version
  raise 'Subclass should supply version
end
```

Then in the subclass
```
class BrakemanScanner < Base
def version
  Gem.loaded_specs["brakeman"].version.to_s
end
end

```
(or self.version if you want to pull these from the class )

This would help encapsulate all the subclasses logic in the subclass (and minimize the number of places we have scanner specific code).

Reply:
@joshua-ostrom That's a very good idea.  I'm totally fine with implementing what you suggested.

A related issue is that we currently hard-code the scanner versions in the README. Perhaps we could automate that too.

@cbtoni , what do you think?

Reply:
Seems fine. I'm not sure its worth trying to automate that in the readme though.

Reply:
@joshua-ostrom , @cbtoni .  
I'm going to implement Josh's idea of adding a `version` method to each scanner.  But instead of adding 
```
def version
  raise 'Subclass should supply version
end
``` 
to the base scanner, I think it makes sense to make the base scanner return empty string.  This is because some scanners have no versions, like ReportNotEmpty.  It's also hard to get versions from some scanners, like DataTheorem, because we do not have direct access to the version.  So for these scanners, it seems good to use empty string as their versions from the base scanner.

Please let me know what you think.

Reply:
Yeah if we don't want to enforce that scanners supply a version then defaulting in base is fine to me